File: catalog/backends/dotnet-aspnetcore/patterns/UnindexedQueryMongoRepository.cs
using MongoDB.Bson;
using MongoDB.Driver;

// Pattern: DBM unindexed query — collection scan (COLLSCAN)
// Adapt: replace collection and field names with domain entities
public class UnindexedQueryMongoRepository
{
    private readonly IMongoCollection<BsonDocument> _events;

    public UnindexedQueryMongoRepository(IMongoClient client)
    {
        _events = client.GetDatabase("demo").GetCollection<BsonDocument>("events");
    }

    public async Task<List<BsonDocument>> FindByCategoryAsync(string category)
    {
        var filter = Builders<BsonDocument>.Filter.Eq("category", category);
        var sort = Builders<BsonDocument>.Sort.Descending("created_at");
        return await _events.Find(filter).Sort(sort).Limit(50).ToListAsync();
    }
}

File: catalog/backends/java-spring/patterns/UnindexedQueryMongoRepository.java
package com.example.service;

import com.mongodb.client.*;
import com.mongodb.client.model.Sorts;
import org.bson.Document;
import java.util.*;

// Pattern: DBM unindexed query — collection scan (COLLSCAN)
// Adapt: replace collection and field names with domain entities
public class UnindexedQueryMongoRepository {

    private final MongoCollection<Document> events;

    public UnindexedQueryMongoRepository(MongoClient client) {
        this.events = client.getDatabase("demo").getCollection("events");
    }

    public List<Document> findByCategory(String category) {
        return events.find(new Document("category", category))
            .sort(Sorts.descending("created_at"))
            .limit(50)
            .into(new ArrayList<>());
    }
}
